flink-metrics/flink-metrics-datadog/src/main/java/org/apache/flink/metrics/datadog/DatadogHttpReporter.java

```diff
@@ -117,7 +117,7 @@ public void close() {
 	@Override
 	public void report() {
 		DatadogHttpRequest request = new DatadogHttpRequest();
-
+		List<Gauge> notNumberGauges = new ArrayList<>();
```

Comment:
while the common use-case a gauge throwing an arbitrary exception also ends up in this list, even if it works with Numbers.

Rename to `gaugesToRemove`.